src/hotspot/share/code/relocInfo.hpp

```diff
@@ -793,8 +850,20 @@ class Relocation {
   int      format()       const { return binding()->format(); }
 
  public:
+  // Make a filler relocation.
+  Relocation() : Relocation(relocInfo::none) {}
+
+  // Intentionally public non-virtual destructor, even though polymorphic.
+  // We never heap allocate a Relocation, so never delete through a base pointer.
+  // RelocationHolder depends on (and verifies) the destructor for all relocation
+  // types is trivial, so can't be virtual.
```

Comment:
Should this be:
```suggestion
  // types is trivial, so can be non-virtual.
```
?

Reply:
Ah, ok. Thanks for explaining.